upgrade_policy/quorum_upgrade/sources/quorum_upgrade_policy.move

```diff
@@ -249,27 +259,35 @@ module quorum_upgrade_policy::quorum_upgrade_policy {
         digest: vector<u8>,
         ctx: &mut TxContext,
     ) {
-        let cap_id = object::id(cap);
-        let proposal_uid = object::new(ctx);
-        let proposal_id = object::uid_to_inner(&proposal_uid);
-        
-        let proposer = tx_context::sender(ctx);
-        
-        event::emit(UpgradeProposed {
-            upgrade_cap: cap_id,
-            proposal: proposal_id,
-            digest,
-            proposer,
-            voters: cap.voters,
-        });
+        transfer::share_object(internal_propose_upgrade(cap, digest, ctx))
+    }
 
-        transfer::share_object(ProposedUpgrade {
-            id: proposal_uid,
-            upgrade_cap: cap_id,
-            proposer,
-            digest,
-            current_voters: vec_set::empty(),
-        })
+    /// V2 of propose_upgrade, returns ProposedUpgrade object which can be used
+    /// in add_metadata function to optionally add metadata.
+    /// Must be used to call share_upgrade_object to share proposal with voters.
+    public fun propose_upgrade_v2(
```

Comment:
why not having a `propose_upgrade_with_metadata`?
personally not a big fun of names that add the version and not clear about the 3 steps needed.
Any reason for that? (I am sure there is one but I cannot think of it)

Reply:
to be honest it does not make a lot of sense!
I am also not sure what the relationship to another API is.
You can add any API you want in the future, I do not understand the correlation. We made metadata a one shot that you cannot override so I do not see the advantage of multiple APIs.
In fact, I just see the annoyance of have to call 2 functions for nothing. And in fact, the way it is, you are calling 3 functions for the most obvious case.
That also adds the fact that the object is now returned instead of being shared in the first place and you have this odd API in `shared_object` that you are supposed to call. 
So, if you ask me, between one obvious function call that does the right thing in "one shot and no mistake", and 3 functions that you have to understand and call properly there is no question whatsoever to what is more proper.
But it's just me of course....

Reply:
and also come to think of it more, with the API the way it is today anybody can add metadata after the fact which to me seems very wrong and possibly risky.
Isn't that the case?

Reply:
@dariorussi maybe we could take this on a call to discuss further with more examples?

For the second argument, it's not the case, only the proposer can add metadata (there's a check there).

Reply:
> only the proposer can add metadata (there's a check there)
fair on the proposer which to be honest does not change much....
we can definitely take this off line, and have that conversation next week, I assume we have not deployed this yet so we can change as we want.
The more I think about this the more I am at a loss on the API we have built to be honest.
But we'll talk more with Tony and you.

See what you have done - lol, you made me think about it now....